src/plugins/intel_gpu/src/plugin/plugin.cpp

```diff
@@ -968,40 +974,101 @@ Parameter Plugin::GetMetric(const std::string& name, const std::map<std::string,
+                    GPU_DEBUG_IF(debug_config->verbose >= 2) {
+                        GPU_DEBUG_COUT << "[MAX_BATCH_SIZE] does not support dynamic networks" << std::endl;
+                    }
+                    return decltype(ov::max_batch_size)::value_type {static_cast<uint32_t>(max_batch_size)};
                 }
-            } else {
-                base_batch_size = 1;
-                GPU_DEBUG_IF(debug_config->verbose >= 1) {
-                    GPU_DEBUG_COUT << "[GPU_MAX_BATCH_SIZE] Batch dimension is not used in inputs." << std::endl;
+                // check the batch dim: either 0th (and the original batch size of 1) or none
+                if (shape.size() && ov::DimensionTracker::get_label(shape[0])) {
+                    const auto& static_shape = input->get_shape();
+                    if (static_shape[0] != 1) {
+                        GPU_DEBUG_IF(debug_config->verbose >= 2) {
+                            GPU_DEBUG_COUT << "[MAX_BATCH_SIZE] Only to be applied for not-batched networks" << std::endl;
+                        }
+                        return decltype(ov::max_batch_size)::value_type {static_cast<uint32_t>(max_batch_size)};
+                    }
+                    batched_inputs.insert(
+                            ngraph::op::util::get_ie_output_name(params[input_id]->output(0)));  // batched dim for the input
+                } else {
+                    // if the 0-th dim is not for the batch, then we support only the case when NONE dimension is batch
+                    for (size_t s = 1; s < shape.size(); s++) {
+                        if (ov::DimensionTracker::get_label(shape[s])) {
+                            GPU_DEBUG_IF(debug_config->verbose >= 2) {
+                                GPU_DEBUG_COUT << "[MAX_BATCH_SIZE] only networks with inputs/outputs batched by 0th dimension" << std::endl;
+                            }
+                            return decltype(ov::max_batch_size)::value_type {static_cast<uint32_t>(max_batch_size)};
+                        }
+                    }
+                }
+            }
+            const auto& results = function->get_results();
+            for (size_t output_id = 0; output_id < results.size(); output_id++) {
+                const auto& output = results[output_id];
+                const auto& shape = output->get_output_partial_shape(0);
+                if (shape.is_dynamic()) {
+                    GPU_DEBUG_IF(debug_config->verbose >= 2) {
+                         GPU_DEBUG_COUT << "[MAX_BATCH_SIZE] does not support dynamic networks." << std::endl;
+                    }
+                    return decltype(ov::max_batch_size)::value_type {static_cast<uint32_t>(max_batch_size)};
+                }
+                // check the batch dim: either 0th (and the original batch size of 1) or none
+                if (shape.size() && ov::DimensionTracker::get_label(shape[0])) {
+                    if (shape[0] != 1) {
```

Comment:
this is particular limitation of the auto-batching (that can batch only originally batch1 networks), not sure the MAX_BATCH_SIZE should report the val;ue of 1 for the originally batched networks

Reply:
I just did same as what is done in auto batch, becuase I thought it was the requirements from the auto plugin's view. For example, the shape is CN, what if MAX_BATCH_SIZE reports 8 while auto batch reports 1? @songbell  Is it allowed?

Reply:
From auto point of view, it's reasonable that GPU plugin report same optimal batch size as batch plugin. 
I think we are talking about the case of the originally batched networks here, auto batch will not re-batch this kind of network, and thus report 1 from batch plugin level. if gpu does not report 1, auto will create more infer request than it actually should do, but if gpu report 1, what is the impact of this?

Reply:
> From auto point of view, it's reasonable that GPU plugin report same optimal batch size as batch plugin. 
This is probably reasonabe from the "OPTIMAL_BATCH_SIZE", but here we are discussing the MAX_BATCH_SIZE.
The MAX_BATCH_SIZE  does NOT depend on whether the auto-batching (or. e.g. a manuyal batching) is used.

> I just did same as what is done in auto batch, becuase I thought it was the requirements from the auto plugin's view. 
The GPU should report the MAX_BATCH_SIZE, and let the auto-batching (when involved) to apply additional limitations. 
The request is rather to do the right or "legal" reshaping (it is unsafe to use the layout). That would also help to align the batch size predications.

Reply:
>> The GPU should report the MAX_BATCH_SIZE, and let the auto-batching (when involved) to apply additional limitations. (via OPTIMAL_BATCH_SIZE).

I think it make sense. I'll update soon

Reply:
@myshevts  Updated. Please review again!